simulation_ws/src/sagemaker_rl_agent/markov/single_machine_training_worker.py

```diff
@@ -40,6 +41,16 @@ def add_items_to_dict(target_dict, source_dict):
 def should_stop_training_based_on_evaluation():
     return False
 
+class SigTermHandler:
+    def __init__(self, data_store):
+        self.data_store = data_store
+        # signal.signal(signal.SIGINT, self.sigterm_exit)
+        signal.signal(signal.SIGTERM, self.sigterm_exit)
+
+    def sigterm_exit(self, signum, frame):
+        self.data_store.unlock()
```

Comment:
This doesn't seem safe to me. Theoretically the lock is there to prevent anyone from trying to read the model until it has been fully written. If the program receives an sigterm then we can't just unlock things and stop the program as we might be in the middle of a write and we'd be corrupting the model. 

This also doesn't help us when the program crashes or is killed without the chance to shutdown gracefully (ie. kill -9). I think we need better recovery logic on startup instead of just removing the lock file. For instance, if you start up and the lock file still exists then maybe you can remove the latest version of the model, which we don't know was successfully written or not, and restore the last known good one. Just removing the lock files seems like it is putting us in a position to have different exceptions with corrupted data.

Reply:
After speaking with @r-meagher, receiving a SIGTERM is expected during normal operation, and removing the `.lock` is something we should do as part of normal operation. The `.lock` file is also only useful during distributed training, and during other times it's more like a vestigial appendix.